### ISSUE TITLE
New "if let" syntax produces errors

### ISSUE THREAD

**Issue by michaellenaghan** [apple/swift-format#409]
```
$ swift-format --version
0.50700.0
```

Here are some examples.

This error:

```
/Users/.../DateTime.swift:234:9: error: file contains invalid or unrecognized Swift syntax.
```

from this code:

```swift
    public init?(floatingDate: Date?, floatingTime: Date? = nil) {
        if let floatingDate {
            ...
```

This error:

```
/Users/.../MetaInfo.swift:70:13: error: file contains invalid or unrecognized Swift syntax.
```

from this code:

```swift
    public var didDeleteOn: Date? {
        get {
            ...
        }
        set {
            if let newValue {
                ...
```

(Xcode compiles both without complaint.)

**Comment by allevato** [comment]
This was a bug in the parser/swift-syntax, so it will need a new release of that package that includes https://github.com/apple/swift-syntax/pull/832 to fix.

**Comment by shimastripe** [comment]
apple/swift-syntax@0.50700.1 is released! and I checked this problem was fixed, please release new swift-format 🙏

**Comment by allevato** [comment]
While a new release isn't strictly necessary here (wiping your build directory and Package.resolved file would automatically pull down the newer swift-syntax revision), I went ahead and released a 0.50700.1 of swift-format that pins to swift-syntax 0.50700.1 (or higher) to make things a bit easier for folks. https://github.com/apple/swift-format/releases/tag/0.50700.1
